internal/cli/cmd/config.go

```diff
@@ -86,6 +88,20 @@ func newCmdConfigSet() *cobra.Command {
 			if err := check.Set(context.Background(), args[0], args[1]); err != nil {
 				fatalf("Unable to set config:  %s", err)
 			}
+			if !params.Restart {
+				fmt.Println("⚠️  Restart Cilium pods for configmap changes to take effect")
+				return nil
+			}
+			if err := k8sClient.DeletePodCollection(
```

Comment:
This should be moved into `Set()` instead as the goal is to make the entire logic by the CLI accessible via a Go API as well.

Reply:
I'm not sure which Go API you are referring here. ` Set()` is currently called only from newCmdConfigSet, which is the handler that backs the CLI command `set config`. Are you suggesting that `Set()` can also be called from some other API in future?